src/Codata/Thunk.agda

```diff
@@ -26,6 +26,14 @@ Thunk^R : ∀ {f g r} {F : Size → Set f} {G : Size → Set g}
           (i : Size) (tf : Thunk F ∞) (tg : Thunk G ∞) → Set r
 Thunk^R R i tf tg = Thunk (λ i → R i (tf .force) (tg .force)) i
 
+------------------------------------------------------------------------
+-- Syntax
+
+Thunk-syntax : ∀ {ℓ} → (Size → Set ℓ) → Size → Set ℓ
+Thunk-syntax = Thunk
+
+syntax Thunk-syntax (λ i → e) j = Thunk[ i < j ] e
```

Comment:
This confuses me: we usually call the larger size `i` and the smaller one `j`. Other than that, I like it!